src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/CaseDataTag.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.digital.ho.hocs.casework.domain.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.PrePersist;
+import javax.persistence.Table;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.UUID;
+
+@Entity
+@Table(name = "case_data_tag")
+@IdClass(CaseDataTagId.class)
+public class CaseDataTag implements Serializable {
```

Comment:
What would be the behaviour in a scenario where a tag is added, deleted, then added again? Do we need a differentiator between the two records stored in the table other than created_at and deleted_on?

Reply:
This doesn't work at all for tags that already exist and have been deleted, good catch...

Reply:
I have altered this to work for deletes and re-adds